src/main/java/app/coronawarn/verification/controller/ExternalTanController.java

```diff
@@ -133,25 +133,13 @@ public DeferredResult<ResponseEntity<Tan>> generateTan(@Valid @RequestBody Regis
       if (appSession.getTanCounter() < tancountermax) {
         AppSessionSourceOfTrust appSessionSourceOfTrust = appSession.getSourceOfTrust();
         TanSourceOfTrust tanSourceOfTrust = TanSourceOfTrust.CONNECTED_LAB;
-        switch (appSessionSourceOfTrust) {
-          case HASHED_GUID:
-            TestResult covidTestResult = testResultServerService.result(new HashedGuid(appSession.getHashedGuid()));
-            if (covidTestResult.getTestResult() != LabTestResult.POSITIVE.getTestResult()
-              && covidTestResult.getTestResult() != LabTestResult.QUICK_POSITIVE.getTestResult()
-            ) {
-              stopWatch.stop();
-              throw new VerificationServerException(HttpStatus.BAD_REQUEST,
-                "Tan cannot be created, caused by the non positive result of the labserver");
-            }
-            break;
-          case TELETAN:
-            tanSourceOfTrust = TanSourceOfTrust.TELETAN;
-            break;
-          default:
-            stopWatch.stop();
-            throw new VerificationServerException(HttpStatus.BAD_REQUEST,
-              "Unknown source of trust inside the appsession for the registration token");
-        }
+
+        TanVerifierFactory tanVerifierFactory = new TanVerifierFactoryImpl();
```

Comment:
Use Spring Dependency Injection

Reply:
Hello @f11h ,

Thank you very much for your feedback! I appreciate it.
I have pushed the changes into the branch.